src/puppetlabs/trapperkeeper/services/metrics/jolokia.clj

```diff
@@ -0,0 +1,78 @@
+(ns puppetlabs.trapperkeeper.services.metrics.jolokia
+  "Clojure helpers for constructing and configuring Jolokia servlets."
+  (:require [clojure.tools.logging :as log]
+            [clojure.walk :as walk]
+            [schema.core :as schema])
+  (:import [org.jolokia.config ConfigKey]
+           [org.jolokia.util LogHandler]
+           [org.jolokia.http AgentServlet]))
+
+
+(def config-mapping
+  "Inspects the Jolokia ConfigKey Enum and generates a mapping that associates
+  a Clojure keyword with each configuration parameter. The keyword used is the
+  camel-cased identifier that would be used to configure a servlet via a
+  web.xml file. For example, `ConfigKey/AGENT_ID` is associated with the
+  keyword `:agentId`.
+
+  For a complete list of configuration options, see:
+
+    https://jolokia.org/reference/html/agents.html#agent-war-init-params"
+  (->> (ConfigKey/values)
+       (map (juxt
+              #(-> % .getKeyValue keyword)
+              identity))
+       (into {})))
+
+(schema/defschema JolokiaConfig
+  "Schema for validating Clojure maps containing Jolokia configuration.
+
+  Creates a map of optional keys which have string values using the
+  config-mapping extracted from the ConfigKey enum."
+  (->> (keys config-mapping)
+       (map #(vector (schema/optional-key %) schema/Str))
+       (into {})))
+
+(def config-defaults
+  "Default configuration values for Jolokia."
+  {;; Enables info logged at debug level.
+   :debug "true"
+   ;; Don't include backtraces in error results returned by the API.
+   :allowErrorDetails "false"
+   ;; Load access policy from: resources/jolokia-access.xml
+   :policyLocation "classpath:/jolokia-access.xml"
+   :mimeType "application/json"})
+
+
+(defn create-servlet-config
+  "Generate Jolokia AgentServlet configuration from a Clojure map"
+  ([]
+   (create-servlet-config {}))
+
```

Comment:
ridiculous nitpick: we don't usually put newlines between different arities of a multi-arity function.